addons/mail/static/src/core/common/mail_fullscreen.js

```diff
@@ -0,0 +1,93 @@
+import { Component, reactive } from "@odoo/owl";
+import { useService } from "@web/core/utils/hooks";
+import { registry } from "@web/core/registry";
+
+const DEFAULT_ID = Symbol("default");
+
+export class MailFullscreen extends Component {
+    static props = [];
+    static template = "mail.Fullscreen";
+
+    setup() {
+        super.setup();
+        this.fullscreen = useService("mail.fullscreen");
+    }
+}
+
+export const fullscreenService = {
+    start() {
+        registry.category("main_components").add("mail.fullscreen", { Component: MailFullscreen });
+        const state = reactive({
+            component: undefined,
+            props: undefined,
+            id: undefined,
+            exit,
+            enter,
+        });
+        async function exit(id = state.id) {
+            if (id !== state.id) {
+                return;
+            }
+            state.component = undefined;
+            state.props = undefined;
+            state.id = undefined;
+            const fullscreenElement =
+                document.webkitFullscreenElement || document.fullscreenElement;
+            if (fullscreenElement) {
+                if (document.exitFullscreen) {
+                    await document.exitFullscreen();
+                } else if (document.mozCancelFullScreen) {
+                    await document.mozCancelFullScreen();
+                } else if (document.webkitCancelFullScreen) {
+                    await document.webkitCancelFullScreen();
+                }
+            }
+        }
+        /**
+         * @param component
+
+         * @param {object} [options]
+         * @param [options.props]
+         * @param {boolean} [options.useNativeFullscreen=true] true if we want to use the device's native fullscreen feature,
```

Comment:
Don't think it's worth adding `useNativeFullscreen = false` feature if unused. I wouldn't mind if there's at least a test but without a test nor a practical use this feels off to have code for that now.